modules/api/src/main/java/com/intuit/wasabi/api/AssignmentsResource.java

```diff
@@ -152,18 +155,18 @@ public Response getAssignment(
         Assignment assignment = getAssignment(userID, applicationName, experimentLabel, context, createAssignment,
                 ignoreSamplingPercent, null, headers);
 
-        return httpHeader.headers().entity(toMap(assignment)).build();
+        return httpHeader.headers().entity(toSingleAssignmentResponseMap(assignment)).build();
     }
 
     private Assignment getAssignment(final User.ID userID, final Application.Name applicationName,
                                      final Experiment.Label experimentLabel, final Context context,
                                      final boolean createAssignment, final boolean ignoreSamplingPercent,
                                      final SegmentationProfile segmentationProfile, final HttpHeaders headers) {
-        Assignment assignment = assignments.getSingleAssignment(userID, applicationName, experimentLabel, context,
-                createAssignment, ignoreSamplingPercent, segmentationProfile, headers, null);
+        Assignment assignment = assignments.doSingleAssignment(userID, applicationName, experimentLabel, context,
```

Comment:
Cosmetic: Can we do away with having so many method parameters. Too many method parameter creates confusion in the code. Can we instead wrap these parameters in a POJO and pass the POJO around instead?